apiserver/common/environ_config.go

```diff
@@ -24,3 +27,23 @@ func (f EnvironConfigGetterFuncs) ModelConfig() (*config.Config, error) {
 func (f EnvironConfigGetterFuncs) CloudSpec() (environs.CloudSpec, error) {
 	return f.CloudSpecFunc()
 }
+
+// NewEnvironFunc is a function that returns a BootstrapEnviron instance.
+type NewEnvironFunc func() (environs.BootstrapEnviron, error)
+
+// EnvironFuncForModel is a helper function that returns a NewEnvironFunc suitable for
+// the specified model.
+func EnvironFuncForModel(model stateenvirons.Model, configGetter environs.EnvironConfigGetter) NewEnvironFunc {
+	var newEnviron NewEnvironFunc
+	if model.Type() == state.ModelTypeCAAS {
+		newEnviron = func() (environs.BootstrapEnviron, error) {
+			f := stateenvirons.GetNewCAASBrokerFunc(caas.New)
+			return f(model)
+		}
+	} else {
+		newEnviron = func() (environs.BootstrapEnviron, error) {
```

Comment:
This would look better if pulled out of the else block and turned into a `return func(...){}`